### DISCUSSION TITLE
Next.js 9.2, Improved Code-Splitting Strategy, Chunk Naming

### DISCUSSION THREAD

**Discussion by roman-may** [vercel/next.js#10707] [Help]
How are the chunk names generated? 
We get names like `chunks/69f05f253a119e56312b77b1ebdfd6f650527a82.8dae1d.js`, where the last part (after the . ) seems to be a content hash and the rest look random. So we get new file names for each build even if there were no changes and wondered if this is the default behavior and if we can change this to hash only to improve cachability?

**Comment by timneutkens** [comment]
webpack is not deterministic in how it generates module ids currently so there's cases where the module numbers could change causing the hash to update.

**Reply by roman-may** [reply]
Thanks for your fast reply!

I'm still think that this is only partly the reason.

We have checked the files before and did find multiple which only differed in some numbers which probably are the webpack module ids that you mentioned. In this case, the file names are completely different between the builds like:
`1eac2115df33ffd300ad309f9ca871bb9f21493e.e152024c020ece611e6a.js`
`f9f39d969bcf68debbe3550c1e0c7f4be476007f.6aa923419661e79eb755.js`

Still, a lot of them had exactly the same content, but they still got new names, but in this case the last part of the name seems to be the content hash and stays the same between the builds.
`7130f7c50038752604262b33e2d117f5aa3080e8.ddcd2236c504ba77a729.js`
`ca40b53671040ded15bc6bbf32a5ef7770d62c15.ddcd2236c504ba77a729.js`

I don't see a reason why it should not be possible to just remove the first part of the name everywhere where its needed. I understand that this is not possible for, for example the pages because of the routing as you explained very good in https://github.com/zeit/next.js/issues/6303#issuecomment-464032996, but this seems to be a completely different case.

**Reply by roman-may** [reply]
Do you know why the first part (before the dot) is needed?
